pkg/testrunner/result/status-uploader.go

```diff
@@ -0,0 +1,113 @@
+package result
+
+import (
+	"context"
+	"fmt"
+	"github.com/gardener/test-infra/pkg/testrunner"
+	"github.com/gardener/test-infra/pkg/testrunner/componentdescriptor"
+	"github.com/gardener/test-infra/pkg/util"
+	"github.com/go-logr/logr"
+	"github.com/google/go-github/v27/github"
+	"github.com/pkg/errors"
+	"io/ioutil"
+	"net/url"
+	"os"
+	"strings"
+
+	"github.com/gardener/gardener/pkg/client/kubernetes"
+
+	tmv1beta1 "github.com/gardener/test-infra/pkg/apis/testmachinery/v1beta1"
+)
+
+// uploads status results as asset to the component
+func UploadStatusToGithub(run *testrunner.Run, component *componentdescriptor.Component, githubUser, githubPassword string) error {
+	tr := run.Testrun
+	md := run.Metadata
+	tableString := strings.Builder{}
+	util.RenderStatusTable(&tableString, tr.Status.Steps)
+	statusOutput := fmt.Sprintf("Testrun: %s\n%s\n%s", tr.GenerateName, tableString.String(), util.PrettyPrintStruct(tr.Status))
+	filename := fmt.Sprintf("%s-%s-%s.txt", md.Landscape, md.CloudProvider, md.KubernetesVersion)
+	if err := ioutil.WriteFile(filename, []byte(statusOutput), 0644); err != nil {
```

Comment:
it seems weird that we need to write the file first then reopen to upload it.
I think we should use `NewUploadRequest` so we can directly pass the `io.Reader`

Reply:
Can you elaborate?

Reply:
To use NewUploadRequest you need the repository client service, which is private unfortunately. Seems like it's not possible to use NewUploadRequest like you suggested. What do you think?

Reply:
Its private for the repositories struct but you have direct access to the client as it is the default `github.Client`.
So just use `client.NewUploadReques()` instead of `client.Repositories.NewUploadRequest `